model/infinispan/src/main/java/org/keycloak/connections/infinispan/DefaultInfinispanConnectionProviderFactory.java

```diff
@@ -209,6 +217,28 @@ protected void initEmbedded() {
         gcb.serialization().marshaller(new JBossUserMarshaller());
 
         cacheManager = new DefaultCacheManager(gcb.build());
+        if (useKeycloakTimeService) {
+            replaceComponent(cacheManager, TimeService.class, new EmbeddedTimeService() {
```

Comment:
We already have KeycloakTestTimeService . How about moving it to `model/infinispan` to avoid using same code on multiple places?

Same probably applies to `DefaultInfinispanConnectionProviderFactory.replaceComponent` which is same code as `InfinispanTestUtil.replaceComponent` . This code uses infinispan internal stuff and it is a chance that during infinispan upgrade to newer version, there will be a need to change this code. So if we have same code with infinispan internals on two places in Keycloak codebase, it possibly  increases the work for the infinispan upgrade in the future IMO.

Or just simplify the testsuite - maybe we can remove `InfinispanTestUtil` entirely and change `TestingResourceProvider.setTestingInfinispanTimeService` and  `TestingResourceProvider.revertTestingInfinispanTimeService` to call `DefaultInfinispanConnectionProviderFactory.setUseKeycloakTimeService` or something like that? WDYT?